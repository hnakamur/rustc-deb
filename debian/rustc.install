usr/bin/rustc
usr/bin/rustdoc
