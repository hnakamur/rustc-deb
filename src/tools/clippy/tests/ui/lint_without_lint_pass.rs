#![deny(clippy::internal)]
#![feature(rustc_private)]

#[macro_use]
extern crate rustc;
use rustc::lint;

#[macro_use]
extern crate clippy_lints;

declare_clippy_lint! {
    pub TEST_LINT,
    correctness,
    ""
}

declare_clippy_lint! {
    pub TEST_LINT_REGISTERED,
    correctness,
    ""
}

pub struct Pass;
impl lint::LintPass for Pass {
    fn get_lints(&self) -> lint::LintArray {
        lint_array!(TEST_LINT_REGISTERED)
    }
}

fn main() {}
