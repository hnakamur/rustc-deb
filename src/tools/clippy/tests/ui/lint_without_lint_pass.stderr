error: the lint `TEST_LINT` is not added to any `LintPass`
  --> $DIR/lint_without_lint_pass.rs:11:1
   |
LL | / declare_clippy_lint! {
LL | |     pub TEST_LINT,
LL | |     correctness,
LL | |     ""
LL | | }
   | |_^
   |
note: lint level defined here
  --> $DIR/lint_without_lint_pass.rs:1:9
   |
LL | #![deny(clippy::internal)]
   |         ^^^^^^^^^^^^^^^^
   = note: #[deny(clippy::lint_without_lint_pass)] implied by #[deny(clippy::internal)]
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

